clang/lib/StaticAnalyzer/Core/LoopUnrolling.cpp

```diff
@@ -84,15 +84,17 @@ ProgramStateRef processLoopEnd(const Stmt *LoopStmt, ProgramStateRef State) {
 
 static internal::Matcher<Stmt> simpleCondition(StringRef BindName,
                                                StringRef RefName) {
+  auto LoopVariable = ignoringParenImpCasts(
+      declRefExpr(to(varDecl(hasType(isInteger())).bind(BindName)))
+          .bind(RefName));
+  auto UpperBound = ignoringParenImpCasts(expr().bind("boundNum"));
```

Comment:
Could we run into any problems if the expression did not have an integer type?

E.g. one could have:
```
for (... ; loopVar < MyWeirdType{}; ...) ;
```

Where `MyWeirdType` has an overloaded `operator<`. Maybe it is fine as the overloaded operators would not be found by the matchers below.

Reply:
WDYT of this one https://github.com/llvm/llvm-project/pull/169400/commits/a7769ed2fa412b8dfab3897a10450f81a2e74aae

I had to apply some refactors to not confuse name lookups (the ambiguities were brought by the AST_MATCHER macro), which I didn't quite like.